### PR TITLE
translation: Update chapter_computational_complexity/performance_evaluation.md

### PR DESCRIPTION
**Pull request by ymmmas**
I've added some parts to align with the Chinese version and rephrased certain sentences for better readability. I personally prefer the term 'Practical Testing' over 'Actual Testing' because it is more commonly used and more relevant to the content. However, I'm not certain about the definitive correct term, so didn’t make the change.

If this pull request (PR) pertains to **Chinese-to-English translation**, please confirm that you have read the contribution guidelines and complete the checklist below:

- [x] This PR represents the translation of a single, complete document, or contains only bug fixes.
- [x] The translation accurately conveys the original meaning and intent of the Chinese version. If deviations exist, I have provided explanatory comments to clarify the reasons.

If this pull request (PR) is associated with **coding or code transpilation**, please attach the relevant console outputs to the PR and complete the following checklist:

- [ ] I have thoroughly reviewed the code, focusing on its formatting, comments, indentation, and file headers.
- [ ] I have confirmed that the code execution outputs are consistent with those produced by the reference code (Python or Java).
- [ ] The code is designed to be compatible on standard operating systems, including Windows, macOS, and Ubuntu.

### CODE REVIEW

File: en/docs/chapter_computational_complexity/performance_evaluation.md [hunk 1]
```diff
@@ -16,11 +16,11 @@ There are mainly two methods of efficiency assessment: actual testing and theore
...
 
-On the other hand, **conducting a full test is very resource-intensive**. As the volume of input data changes, the efficiency of the algorithms may vary. For example, with smaller data volumes, algorithm `A` might run faster than `B`, but the opposite might be true with larger data volumes. Therefore, to draw convincing conclusions, we need to test a wide range of input data sizes, which requires significant computational resources.
+On the other hand, **conducting a full test is very resource-intensive**. As the volume of input data changes, the efficiency of the algorithms also changes. For example, with smaller data volumes, algorithm `A` might run faster than `B`, but with larger data volumes, the test results may be the opposite. Therefore, to draw convincing conclusions, we need to test a wide range of input data sizes, which requires excessive computational resources.
```
Comment (SamJin98) [inline en/docs/chapter_computational_complexity/performance_evaluation.md:23]: suggestion:
'As the volume of input data changes, the efficiency of the algorithms also changes.'
to
'Algorithm efficiency varies with input data size.'

File: en/docs/chapter_computational_complexity/performance_evaluation.md [hunk 2]
```diff
@@ -16,11 +16,11 @@ There are mainly two methods of efficiency assessment: actual testing and theore
...
 
-On one hand, **it's difficult to eliminate interference from the testing environment**. Hardware configurations can affect algorithm performance. For example, algorithm `A` might run faster than `B` on one computer, but the opposite result may occur on another computer with different configurations. This means we would need to test on a variety of machines to calculate average efficiency, which is impractical.
+On one hand, **it's difficult to eliminate interference from the testing environment**. Hardware configurations can affect algorithm performance. For example, an algorithm with a high degree of parallelism is better suited for running on multi-core CPUs, while an algorithm that involves intensive memory operations performs better with high-performance memory. The test results of an algorithm may vary across different machines. This means we would need to test the algorithm on various machines to calculate average efficiency, which is impractical.
```
Comment (SamJin98) [inline en/docs/chapter_computational_complexity/performance_evaluation.md:21]: suggestion:
'...we need to test the algorithm on various machines to calculate average efficiency, which is impractical.'
to
'...testing across multiple machines to calculate average efficiency becomes impractical.'

**Review by SamJin98 — COMMENTED** [review]
minor changes :)

File: en/docs/chapter_computational_complexity/performance_evaluation.md [hunk 3]
```diff
@@ -2,10 +2,10 @@
 
 In algorithm design, we pursue the following two objectives in sequence.
 
-1. **Finding a Solution to the Problem**: The algorithm should reliably find the correct solution within the stipulated range of inputs.
+1. **Finding a Solution to the Problem**: The algorithm should reliably find the correct solution within the specified range of inputs.
 2. **Seeking the Optimal Solution**: For the same problem, multiple solutions might exist, and we aim to find the most efficient algorithm possible.
 
-In other words, under the premise of being able to solve the problem, algorithm efficiency has become the main criterion for evaluating the merits of an algorithm, which includes the following two dimensions.
+In other words, under the premise of being able to solve the problem, algorithm efficiency has become the main criterion for evaluating the quality of an algorithm, which includes the following two dimensions.
```
Comment (yuelinxin) [inline en/docs/chapter_computational_complexity/performance_evaluation.md:8]: it might be better here to just not have anything, as in “for evaluating an algorithm”

**Review by yuelinxin — COMMENTED** [review]
Great work, please consider my comment

File: en/docs/chapter_computational_complexity/performance_evaluation.md (hunk 2 above)
Comment (ymmmas) [reply en/docs/chapter_computational_complexity/performance_evaluation.md]: Thank you for your suggestions, I have revised the translation according to the suggestion!

**Review by krahets — APPROVED** [review]
